x-pack/legacy/plugins/cloud_migration/server/routes/cluster.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import axios from 'axios';
+import { Router, RouterRouteHandler } from '../../../../server/lib/create_router';
+
+import { LocalClusterState } from '../../common/types';
+
+import { Plugins } from '../shim';
+
+const getServerGeoId = async (): Promise<{ country: string; continent: string }> => {
+  // GeoIP the server
+  const axiosOptions = {
+    headers: {
+      'Content-Type': 'application/json',
+      'X-Routing-Key': 'super-secret',
+    },
+    validateStatus: () => true,
+  };
+  const {
+    data: { ip },
+  } = await axios.get('http://ipinfo.io', axiosOptions);
+
+  const {
+    data: { country, continent },
+  } = await axios.get(`http://www.iplocate.io/api/lookup/${ip}`, axiosOptions);
```

Comment:
Will the `/cluster/state` route fail if the ipinfo or iplocate services are unavailable? Is that what we want?

Reply:
This is a quick draft I made to get the server location. The `/cluster/state` route should, of course, be resilient to the geo info lookup failing.

I want to see first if this is something we want, and if we do, if using a public endpoint is OK with the security team. I had to use the `http` scheme as the `https` was blocked by Kibana as not being whitelisted.